src/thunks/searchThunk.ts

```diff
@@ -1,74 +1,99 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
 import { createAsyncThunk } from '@reduxjs/toolkit';
 
 import { treeherderBaseURL } from '../common/constants';
 import type { APIPushResponse } from '../types/api';
-import type { Repository, Revision } from '../types/state';
+import type { Repository, RevisionsList, InputType } from '../types/state';
+
+interface FetchDataArgs {
+  repository: Repository['name'];
+  searchType: InputType;
+}
+
+interface FetchDataArgsIDEmail {
+  repository: Repository['name'];
+  search: string;
+  searchType: InputType;
+}
+
+interface FetchDataMeta {
+  error: string;
+  searchType: InputType;
+}
 
 export const fetchRecentRevisions = createAsyncThunk<
-  Revision[],
-  Repository['name'],
-  { rejectValue: string }
->('search/fetchRecentRevisions', async (repository, { rejectWithValue }) => {
-  let response;
-  try {
-    response = await fetch(
-      `${treeherderBaseURL}/api/project/${repository}/push/?hide_reviewbot_pushes=true`,
-    );
-  } catch (err) {
-    return rejectWithValue((err as Error).message);
-  }
-  const json = (await response.json()) as APIPushResponse;
-  if (json.results.length > 0) {
-    return json.results;
-  }
+  RevisionsList[],
+  FetchDataArgs,
+  { rejectValue: FetchDataMeta }
+>(
+  'search/fetchRecentRevisions',
+  async ({ repository, searchType }, { rejectWithValue }) => {
+    let response;
+
+    try {
+      response = await fetch(
+        `${treeherderBaseURL}/api/project/${repository}/push/?hide_reviewbot_pushes=true`,
+      );
+    } catch (err) {
+      const error = (err as Error).message;
+      return rejectWithValue({ error, searchType });
```

Comment:
See above for the comment that you don't need the searchType in the rejection value. (and this applies below too of course)